packages/presets/src/fragments/copilot-panel/copilot-service/copilot-config.ts

```diff
@@ -24,6 +26,52 @@ type VendorPack<T extends AllServiceKind> = {
   impl: ServiceImpl<GetMethod<T>, unknown>;
 };
 
+/**
+ * A simple PRNG, using a linear congruential generator.
+ * reference link: https://en.wikipedia.org/wiki/Linear_congruential_generator
+ */
+class SimplePRNG {
+  constructor(private seed: number) {}
+
+  next() {
+    const a = 1664525;
+    const c = 1013904223;
+    const m = 2 ** 32;
+    this.seed = (a * this.seed + c) % m;
+    return this.seed / m;
+  }
+}
+
+function shuffleArray(array: number[], seed: number) {
+  const prng = new SimplePRNG(seed);
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(prng.next() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+}
+
+/**
+ * It's used for encoding the OpenAI key in the URL.
```

Comment:
Don't put the sensitive encoding logic (and also don't mention OAI 😅) in this public repo. Instead, you can [use a secret gist](https://chat.openai.com/share/a855a26b-fcf8-41ea-8cd6-c0d452e96321) and share the encode script internally.